masquerade-core/src/main/java/com/flipkart/masquerade/processor/OverrideProcessor.java

```diff
@@ -0,0 +1,200 @@
+        for (Field field : getNonStaticFields(clazz)) {
+            Class<? extends Annotation> annotationClass = rule.getAnnotationClass();
+            Annotation[] annotations = field.getAnnotationsByType(annotationClass);
+            if (annotations != null && annotations.length != 0) {
+                for (Annotation annotation : annotations) {
+                    constructOperation(rule, annotationClass, annotation, methodBuilder, field.getName());
+                }
+            }
+
+            addRecursiveStatement(field, methodBuilder);
+        }
+
+        return generateImplementationType(rule, clazz, implName, methodBuilder.build());
+    }
+
+    /**
+     * @param rule Current Rule
+     * @param clazz Current Class
+     * @return A MethodSpec builder which overrides the interface method
+     */
+    private MethodSpec.Builder generateOverrideMethod(Rule rule, Class<?> clazz) {
+        MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder(INTERFACE_METHOD);
+        methodBuilder.addAnnotation(Override.class);
+        methodBuilder.addModifiers(Modifier.PUBLIC);
+        methodBuilder.addParameter(clazz, OBJECT_PARAMETER);
+        methodBuilder.addParameter(rule.getEvaluatorClass(), EVAL_PARAMETER);
+        methodBuilder.addParameter(getEntryClass(configuration), CLOAK_PARAMETER);
+        return methodBuilder;
+    }
+
+    /**
+     * @param rule Current Rule
+     * @param clazz Current Class
+     * @param implName Name of the implementation class
+     * @param method The overridden method the implementation class
+     * @return Constructed TypeSpec for the implementation class
+     */
+    private TypeSpec generateImplementationType(Rule rule, Class<?> clazz, String implName, MethodSpec method) {
+        TypeSpec.Builder implBuilder = TypeSpec.classBuilder(implName);
+        implBuilder.addModifiers(Modifier.PUBLIC);
+        /* Implements the interface and attaches the current class as a Generic bound */
+        implBuilder.addSuperinterface(ParameterizedTypeName.get(getRuleInterface(configuration, rule), TypeName.get(clazz)));
+        implBuilder.addMethod(method);
+        return implBuilder.build();
+    }
+
+    /**
+     * Adds a call to the Entry class if the current field needs to be processed as well
+     * @param field Current Field
+     * @param methodBuilder Current method builder
+     */
+    private void addRecursiveStatement(Field field, MethodSpec.Builder methodBuilder) {
+        /* Does not add the statement if the field is primitive, primitive wrapper, String or an Enum */
+        if (!field.getType().isPrimitive() &&
+                !getWrapperTypes().contains(field.getType()) &&
+                !String.class.isAssignableFrom(field.getType()) &&
+                !field.getType().isEnum()) {
+            methodBuilder.addStatement("$L.$L($L.$L(), $L)", CLOAK_PARAMETER, ENTRY_METHOD, OBJECT_PARAMETER, getGetterName(field.getName()), EVAL_PARAMETER);
```

Comment:
It would have, I've added an annotation that when used would stop the recursive call